src/unicode.cpp

```diff
@@ -5,22 +5,42 @@ namespace ada::unicode {
   // A forbidden host code point is U+0000 NULL, U+0009 TAB, U+000A LF, U+000D CR, U+0020 SPACE, U+0023 (#),
   // U+002F (/), U+003A (:), U+003C (<), U+003E (>), U+003F (?), U+0040 (@), U+005B ([), U+005C (\), U+005D (]),
   // U+005E (^), or U+007C (|).
-  ada_really_inline constexpr bool is_forbidden_host_code_point(const char c) noexcept {
-    return c == '\0' || c == '\t' || c == '\n' || c == '\r' || c == ' ' ||
-              c == '#' || c == '/' || c == ':' || c == '?' || c == '@' ||
-              c == '[' || c == '<' || c == '>' || c == '\\' || c == ']' ||
-              c == '^' || c == '|';
+  ada_really_inline static bool is_forbidden_host_code_point(const char c) noexcept {
```

Comment:
I believe this can be a `constexpr` right? And can we move this table declaration outside of the function?

Reply:
The following will work but requires care not to mix the table names...

```C++
constexpr static bool table[] = {
       1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0};
constexpr bool is_forbidden_host_code_point2(const char c) noexcept {
     return table[uint8_t(c) & 0x7f];
}

```

Note that constexpr will not improve the performance at runtime.